sdk/identity/azure-identity/src/main/java/com/azure/identity/WorkloadIdentityCredential.java

```diff
@@ -89,6 +92,13 @@ public class WorkloadIdentityCredential implements TokenCredential {
         ClientAssertionCredential tempClientAssertionCredential = null;
         String tempClientId = null;
 
+        if (identityClientOptions.isKubernetesTokenProxyEnabled()) {
+            if (CustomTokenProxyConfiguration.isConfigured(configuration)) {
```

Comment:
we should throw when AZURE_KUBERNETES_TOKEN_PROXY has no value but any of the other variables does. `parseAndValidate` handles that but `isConfigured` returns true only when AZURE_KUBERNETES_TOKEN_PROXY has a value; when it doesn't, we don't call `parseAndValidate` and therefore don't throw

Reply:
+1